server/strategy_test.go

```diff
@@ -135,46 +136,20 @@ func newData() testData {
 		InPlaceUpdate: true,
 	}
 
-	nodeStatuses := make(map[string]*cke.NodeStatus)
-	for _, nodeName := range nodeNames {
-		nodeStatuses[nodeName] = &cke.NodeStatus{
-			Etcd:         cke.EtcdStatus{ServiceStatus: cke.ServiceStatus{Running: false}, HasData: false, IsAddedMember: false},
-			SSHConnected: true,
-		}
-	}
-
-	var nodeList []corev1.Node
-	for _, nodeName := range nodeNames {
-		nodeList = append(nodeList, corev1.Node{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: nodeName,
-			},
-			Status: corev1.NodeStatus{
-				Conditions: []corev1.NodeCondition{
-					{
-						Type:   corev1.NodeReady,
-						Status: corev1.ConditionTrue,
-					},
-				},
-			},
-		})
-	}
-	nodeList[0].Labels = map[string]string{op.CKELabelMaster: "true"}
-	nodeList[1].Labels = map[string]string{op.CKELabelMaster: "true"}
-	nodeList[2].Labels = map[string]string{op.CKELabelMaster: "true"}
-	nodeList[3].Annotations = cluster.Nodes[3].Annotations
-	nodeList[3].Labels = cluster.Nodes[3].Labels
-	nodeList[3].Spec.Taints = cluster.Nodes[3].Taints
-
 	status := &cke.ClusterStatus{
 		ConfigVersion: cke.ConfigVersion,
-		NodeStatuses:  nodeStatuses,
-		Kubernetes: cke.KubernetesClusterStatus{
-			ResourceStatuses: map[string]cke.ResourceStatus{
-				"Namespace/foo": {Annotations: map[string]string{cke.AnnotationResourceRevision: "1"}},
-			},
-			Nodes: nodeList,
-		},
+		// All nodes are ssh connected, but no services are running.
+		NodeStatuses: map[string]*cke.NodeStatus{
+			nodeNames[0]: {SSHConnected: true},
```

Comment:
In real environments, other fields such as `Etcd` would also be filled in by `GetNodeStatus()`, but with the same values as the defaults.
It would be good to describe such implicit assumptions, but...
面倒ですね。